### PR TITLE
Capture CSS in our index

### PR DESCRIPTION
**Pull request by habdelra**
This PR captures CSS in our index as we visit modules during indexing

### CODE REVIEW

File: packages/host/tests/integration/search-index-test.gts [hunk 1]
```diff
@@ -976,6 +976,65 @@ module(`Integration | search-index`, function (hooks) {
...
+            meta: {
+              adoptsFrom: { module: `./person`, name: 'Person' },
+            },
+          },
+        },
+      },
+    });
+
+    let indexer = realm.searchIndex;
+    let scope = `data-scopedcss-a61576b4d4-6ad446b263`;
```
Comment (backspace) [inline packages/host/tests/integration/search-index-test.gts:1014]: This is surely fine for now but we could extract [the composition of this attribute](https://github.com/cardstack/glimmer-scoped-css/blob/main/glimmer-scoped-css/src/ast-transform.ts#L24) if it ends up changing.

File: packages/runtime-common/search-index.ts [hunk 2]
```diff
@@ -235,6 +236,12 @@ export class SearchIndex {
     return { type: 'doc', doc };
   }
 
+  // this is for tests--it's unlikely we'd actually want to access CSS directly
+  // this way
+  async css(url: URL, opts?: Options): Promise<IndexedCSSOrError | undefined> {
```
Comment (backspace) [inline packages/runtime-common/search-index.ts:241]: I couldn’t find this being used anywhere but if it’s test-specific could this package have a test helpers file instead?

**Review by backspace — APPROVED** [review]

File: packages/host/tests/integration/search-index-test.gts [hunk 3]
```diff
@@ -976,6 +976,65 @@ module(`Integration | search-index`, function (hooks) {
...
+    let entry = await indexer.css(new URL(`${testRealmURL}person`));
```
Comment (habdelra) [inline packages/host/tests/integration/search-index-test.gts:1015]: @backspace SearchIndex.css() is used here. I’m unsure how you would be able to extract this into a test helpers file as there is no other way to directly access the indexer’s state.

Comment (backspace) [reply packages/host/tests/integration/search-index-test.gts:1015]: 👍🏻 makes sense, thanks

File: packages/host/app/lib/current-run.ts [hunk 4]
```diff
@@ -308,14 +310,25 @@ export class CurrentRun {
...
-        consumes.map((d) => trimExecutableExtension(new URL(d)).href),
-      ),
+      deps: new Set(deps),
     });
+
+    let request = await this.loaderService.loader.fetch(url.href);
+    let transpiledSrc = await request.text();
+    let css = getScopedCss(transpiledSrc);
+    if (css) {
+      await this.batch.updateEntry(url, {
```
Comment (jurgenwerk) [inline packages/host/app/lib/current-run.ts:325]: So for example if we had `http://test-realm/person` module, am I correct to say there will be 2 index entries with this url, one for with the type `module` and the other one with type `css`?

**Review by jurgenwerk — APPROVED** [review]
Looks good, looking forward to using it in the search api

File: packages/host/app/lib/current-run.ts (hunk 4 above)
Comment (habdelra) [reply packages/host/app/lib/current-run.ts:325]: Yup, that’s exactly right. All css entries will have a corresponding module entry at the same url.

Comment (habdelra) [reply packages/host/app/lib/current-run.ts:325]: Probably when you design the search query, I’d suggest using a different query than the one we are currently using for searching for  instances. In the html search query likely you’ll need to do a lateral cross join against the deps column and join that against all the css with the same file_alias.

**Pull request merged**